include/swift/AST/Types.h

```diff
@@ -1113,6 +1113,21 @@ class alignas(1 << TypeAlignInBits) TypeBase {
   void *operator new(size_t Bytes, void *Mem) throw() { return Mem; }
 };
 
+/// Holds the information needed to print a type for use in a specific context.
+/// Intended for use when printing a type as part of a fix-it or other
+/// diagnostic message where shadowing is a concern.
+struct TypeDescription {
```

Comment:
I'm not all that happy with the name TypeDescription, but couldn't come up with anything better that was reasonable concise.